packages/strapi-plugin-graphql/services/shadow-crud.js

```diff
@@ -257,7 +257,11 @@ const buildAssocResolvers = model => {
             };
 
             if (['oneToOne', 'oneWay', 'manyToOne'].includes(nature)) {
-              if (!_.has(obj, alias) || _.isNil(foreignId)) {
+              if (
+                !_.has(obj, alias) ||
+                _.isNil(foreignId) ||
+                (_.isObject(obj[alias]) && !_.has(obj[alias], targetPK))
```

Comment:
This would work but we should look into fixing the root cause instead (the empty object) Are you up for looking into it ?

Reply:
It appears to be an issue in the base action itself because the data provided by the resolver is that which is returned via the REST API action. As such, the REST API also returns the empty object.

Reply:
This change limits the scope of any issues to graphql users only.

Reply:
After a little longer investigating, I believe it's originating within the `strapi-connector-bookshelf`. I'm not going to look any further.

Reply:
Yes I think we should actually fix the connector as there is no good reason to return an empty object in REST either.